content/apply.js

```diff
@@ -138,30 +152,26 @@ self.INJECTED !== 1 && (() => {
     }
   }
 
-  function doDisableAll(disableAll) {
+  function updateDisableAll(key, disableAll) {
     if (STYLE_VIA_API) {
       API.styleViaAPI({method: 'prefChanged', prefs: {disableAll}});
     } else {
       styleInjector.toggle(!disableAll);
     }
   }
 
-  function fetchParentDomain() {
-    return parentDomain ?
-      Promise.resolve() :
-      API.getTabUrlPrefix()
-        .then(newDomain => {
-          parentDomain = newDomain;
-        });
-  }
-
-  function updateExposeIframes() {
-    if (!prefs.get('exposeIframes') || window === parent || !styleInjector.list.length) {
-      document.documentElement.removeAttribute('stylus-iframe');
+  async function updateExposeIframes(key, value = prefs.get('exposeIframes')) {
+    const attr = 'stylus-iframe';
+    const el = document.documentElement;
+    if (!el) return; // got no styles so styleInjector didn't wait for <html>
+    if (!value || window === parent || !styleInjector.list.length) {
```

Comment:
Maybe we can just avoid registering the listener when `window === parent`.

Reply:
It wasn't subscribed, but we unconditionally ran it anyway in line 49, so yeah makes sense to move it there and simplify the condition here.